### PR TITLE
Refact caps handler

### PR DESCRIPTION
**Pull request by mitya52**

### CODE REVIEW

File: src/caps.rs [hunk 1]
```diff
@@ -75,40 +76,50 @@ pub async fn load_caps(
...
-        headers.insert(client_version_header, reqwest::header::HeaderValue::from_str(build::PKG_VERSION).unwrap());
-        if !api_key.is_empty() {
-            headers.insert(reqwest::header::AUTHORIZATION, reqwest::header::HeaderValue::from_str(format!("Bearer {}", api_key).as_str()).unwrap());
+        let mut status: u16 = 0;
+        for url in caps_urls.iter() {
+            let api_key = cmdline.api_key.clone();
+            let http_client = global_context.read().await.http_client.clone();
+            let mut headers = reqwest::header::HeaderMap::new();
+            let client_version_header = reqwest::header::HeaderName::from_static("client_version");
+            headers.insert(client_version_header, reqwest::header::HeaderValue::from_str(build::PKG_VERSION).unwrap());
```
Comment (valaises) [inline src/caps.rs:111]: don't need this header anymore

File: src/caps.rs [hunk 2]
```diff
@@ -75,40 +76,50 @@ pub async fn load_caps(
...
+            if !api_key.is_empty() {
+                headers.insert(reqwest::header::AUTHORIZATION, reqwest::header::HeaderValue::from_str(format!("Bearer {}", api_key).as_str()).unwrap());
+            }
+            caps_url = url.clone();
+            let response = http_client.get(caps_url.clone()).headers(headers).send().await.map_err(|e| format!("{}", e))?;
```
Comment (valaises) [inline src/caps.rs:116]: &caps_url

File: src/caps.rs [hunk 3]
```diff
@@ -75,40 +76,50 @@ pub async fn load_caps(
...
+            let joined_url_fallback = base_url.join(&CAPS_FILENAME_FALLBACK).map_err(|_| "failed to parse address url (2)".to_string())?;
+            caps_urls.push(joined_url.to_string());
+            caps_urls.push(joined_url_fallback.to_string());
         } else {
             is_local_file = true;
-            caps_url = cmdline.address_url.clone();
+            caps_urls.push(cmdline.address_url.clone());
         }
     }
+    let mut caps_url: String = caps_urls[0].clone();
```
Comment (valaises) [inline src/caps.rs:99]: caps_url = match caps_urls.get(0) {
Some(url) => url,
Err(e) => ...do something with error
}

File: src/caps.rs (hunk 1 above)
Comment (mitya52) [reply src/caps.rs:111]: oops

**Review by valaises — APPROVED** [review]

**Pull request merged**